noteable/sql/sqlalchemy/__init__.py

```diff
@@ -291,8 +311,60 @@ def preprocess_configuration(
         dsn_dict["query"] = query
 
 
+class BasePostgreSQLConnection(IntrospectableSQLAlchemyConnection):
+    needs_explicit_commit = False
+    _installed_psycopg2_interrupt_fix: bool = False
+    schemas_to_avoid = ('pg_catalog', 'information_schema')
+
+    @classmethod
+    def preprocess_configuration(
+        cls, datasource_id: str, dsn_dict: Dict[str, Any], create_engine_kwargs: Dict[str, Any]
+    ) -> None:
+        """Install fix for ENG-4327 (cannot interrupt kernels doing SQL queries)
```

Comment:
Since this is public source code, I recommend not including any JIRA ticket numbers whatsoever. Description inside is good. However it's worth noting that any `__doc__` is a public interface and likely one that gets exposed to ChatGPT.

Reply:
yeah will clean up sooner and later.